491 formal spec reorg (#712)
finos/FDC3: docs/intents/spec.md

* Distinct
    * Granular enough that Intent handlers can communicate key functional differences

## Namespaces
All standard intent names are reserved. Applications may use their own intents ad hoc.
However, there is a need for applications to ensure that their intents avoid collision. The recommended approach here is to use the app name as the noun. For example, the ‘myChart’ App may expose the ‘ViewChart’ intent and the ‘myChart.Foo’ proprietary intent.

## Using Intents
Combined with [context data](../context/overview) and [App Directory](../app-directory/overview) standards, intents enable rich service discovery on the desktop. For example:

### Ask for a chart to be displayed
```javascript
const result = await fdc3.raiseIntent("ViewChart", {

>  type: "fdc3.instrument",
>  name: "IBM",
>  id: {

Reviewer: ```suggestion
  type: "fdc3.instrument",
  name: "IBM",
  id: {
```

    ticker:"ibm"
  }
});
```

### Ask a specific application to display a chart
```javascript
const result = await fdc3.raiseIntent("ViewChart", {

>  type: "fdc3.instrument",
>  name: "IBM",
>  id: {

Reviewer: ```suggestion
  type: "fdc3.instrument",
  name: "IBM",
  id: {
```

    ticker:"ibm"
  }
}, "market-data-app");
```

### Find applications that can start a chat
```javascript
const intentApps = await fdc3.findIntent("StartChat");
```

### Find available intents for a contact
```javascript
const intentsAndApps = await fdc3.findIntentsByContext({
 type: "fdc3.contact",
 name: "Jane Doe",
 id: {
    email:"jane@doe.com"
  }
});
```

## Using Intents without a context
As the [Desktop Agent API](../api/ref/DesktopAgent) and [App Directory](../app-directory/overview) both
require a context to be specified wherever intents are used, using an intent without a context is

> achieved through the use of an explcit `null` context type `fdc3.nothing`. By using an explicit type
> to represent a lack of context we allow applications to declare their support for a lack of 
> context.

Reviewer: ```suggestion
achieved through the use of the explicit `null` context type `fdc3.nothing`. By using an explicit type
to represent an absence of context we allow applications to declare their support for an absence of 
context.
```
